File: Backend/LogisticsPlatform.Application/DTOs/Orders/UpdateOrderDto.cs
using LogisticsPlatform.Domain.Enums;

namespace LogisticsPlatform.Application.DTOs.Orders
{
    public class UpdateOrderDto
    {
        public OrderType? OrderType { get; set; }
        public OrderDirection? Direction { get; set; }
        public OrderDateDto? StartDate { get; set; }
        public OrderDateDto? EndDate { get; set; }
        public LookupValueDto? StartDateType { get; set; }
        public LookupValueDto? EndDateType { get; set; }

        public OrderDateDto? PlannedPickup { get; set; }
        public OrderDateDto? PlannedDelivery { get; set; }

        public string? DispatchNotes { get; set; }
        public string? DeliveryNotes { get; set; }

        public Guid? PreferredCarrierId { get; set; }

        public string? PrimaryPONumber { get; set; }
        public string? PrimaryBolNumber { get; set; }
        public string? PrimaryProNumber { get; set; }

        public string? Commodity { get; set; }
        public decimal? TotalWeight { get; set; }
        public int? TotalPallets { get; set; }
        public decimal? TotalVolume { get; set; }

        public decimal? CustomerRate { get; set; }
    }
}

File: Frontend/src/app/core/models/orders/update-order.dto.ts
export interface UpdateOrderDto {
    orderType?: string | null;
    direction?: string | null;
    startDate?: { date: string; timezone?: string | null; hasTime: boolean } | null;
    endDate?: { date: string; timezone?: string | null; hasTime: boolean } | null;
    startDateType?: { key?: string | null; value?: string | null } | null;
    endDateType?: { key?: string | null; value?: string | null } | null;

    plannedPickup?: { date: string; timezone?: string | null; hasTime: boolean } | null;
    plannedDelivery?: { date: string; timezone?: string | null; hasTime: boolean } | null;

    dispatchNotes?: string | null;
    deliveryNotes?: string | null;

    preferredCarrierId?: string | null;

    primaryPONumber?: string | null;
    primaryBolNumber?: string | null;
    primaryProNumber?: string | null;
    commodity?: string | null;
    totalWeight?: number | null;
    totalPallets?: number | null;
    totalVolume?: number | null;

    customerRate?: number | null;
}
